ndb_api/db_helpers.go

```diff
@@ -184,62 +210,7 @@ func (a *MSSQLProvisionRequestAppender) appendRequest(req *DatabaseProvisionRequ
 	profileMap := reqData[common.PROFILE_MAP_PARAM].(map[string]ProfileResponse)
 	dbParamInstanceProfile := profileMap[common.PROFILE_TYPE_DATABASE_PARAMETER_INSTANCE]
 
-	actionArgs := []ActionArgument{
-		{
-			Name:  "working_dir",
-			Value: "C:\\temp",
-		},
-		{
-			Name:  "sql_user_name",
-			Value: "sa",
-		},
-		{
-			Name:  "authentication_mode",
-			Value: "windows",
-		},
-		{
-			Name:  "delete_vm_on_failure",
-			Value: "false",
-		},
-		{
-			Name:  "is_gmsa_sql_service_account",
-			Value: "false",
-		},
-		{
-			Name:  "provision_from_backup",
-			Value: "false",
-		},
-		{
-			Name:  "distribute_database_data",
-			Value: "true",
-		},
-		{
-			Name:  "retain_database_in_restoring_mode",
-			Value: "false",
-		},
-		{
-			Name:  "dbserver_name",
-			Value: database.GetDBInstanceName(),
-		},
-		{
-			Name:  "server_collation",
-			Value: "SQL_Latin1_General_CP1_CI_AS",
-		},
-		{
-			Name:  "database_collation",
-			Value: "SQL_Latin1_General_CP1_CI_AS",
-		},
-		{
-			Name:  "dbParameterProfileIdInstance",
-			Value: dbParamInstanceProfile.Id,
-		},
-		{
-			Name:  "vm_dbserver_admin_password",
-			Value: adminPassword,
-		},
-	}
-
-	req.ActionArguments = append(req.ActionArguments, actionArgs...)
+	appendActionArguments(req, getMsSQLOverridableActionArgs(dbParamInstanceProfile.Id, adminPassword), getMsSQLNonOverridableActionArgs(database.GetDBInstanceName()))
```

Comment:
Can this logic be simplified to do the following:
1. Have a default list of action arguments (like earlier).
2. Iterate over the user provided inputs.
3. if name of the arg matches, then replace the value with the user's value
4. Else add the name/value pair.

Reply:
Yes, implemented.